ostelco-ios-client/ViewControllers/Home/BecomeAMemberViewController.swift

```diff
@@ -15,54 +15,81 @@ class BecomeAMemberViewController: ApplePayViewController {
 
     @IBOutlet private weak var buttonContainer: UIView!
 
-    @IBAction private func cancelButton(_ sender: Any) {
-        dismiss(animated: true, completion: nil)
-    }
-    
+    var paymentButton: PKPaymentButton? = nil
+    var plan: Product? = nil
+
     override func viewDidLoad() {
         super.viewDidLoad()
-        
-        // 1. Make sure device supports apple pay and that there are no other restrictions preventing payment (like parental control)
-        if PKPaymentAuthorizationViewController.canMakePayments() {
-            
-            let paymentButton: PKPaymentButton
-            
-            // 2. Check if user has a stripe supported card in its wallet
-            if Stripe.deviceSupportsApplePay() {
-                paymentButton = PKPaymentButton(paymentButtonType: .checkout, paymentButtonStyle: .whiteOutline)
-                paymentButton.addTarget(self, action: #selector(BecomeAMemberViewController.buyButtonTapped), for: .touchUpInside)
-            } else {
-                paymentButton = PKPaymentButton(paymentButtonType: .setUp, paymentButtonStyle: .whiteOutline)
-                paymentButton.addTarget(self, action: #selector(BecomeAMemberViewController.setUpButtonTapped), for: .touchUpInside)
-            }
-            paymentButton.translatesAutoresizingMaskIntoConstraints = false
-            buttonContainer.addSubview(paymentButton)
-            
-            paymentButton.widthAnchor.constraint(equalTo: buttonContainer.widthAnchor).isActive = true
-            paymentButton.heightAnchor.constraint(equalTo: buttonContainer.heightAnchor).isActive = true
-            paymentButton.centerXAnchor.constraint(equalTo: buttonContainer.centerXAnchor).isActive = true
-            paymentButton.centerYAnchor.constraint(equalTo: buttonContainer.centerYAnchor).isActive = true
-            if #available(iOS 12.0, *) {
-                paymentButton.cornerRadius = 8.0
-            }
+        setupPaymentButton()
+        let spinnerView: UIView = showSpinner(onView: view)
+        getProducts { products, error in
+            self.removeSpinner(spinnerView)
+            self.plan = self.getFirstPlan(products)
+        }
+    }
+
+    private func getFirstPlan(_ products: [Product]) -> Product? {
+        // See if the list contains offers.
+        if let firstPlan = products.first(where: { $0.type == "plan" }) {
+            return firstPlan
         }
-        
+        return nil
     }
-    
+
+    func setupPaymentButton() {
+        var showSetupButton = false
+        let applePayError:ApplePayError? = canMakePayments()
```

Comment:
You should be seeing a SwiftLint warning on this line because of the colon spacing - please let me know if you're not.